view/frontend/templates/js/replacejs.phtml

```diff
@@ -640,6 +640,16 @@ ob_start();
         // Create Bolt order and configure BoltCheckout
         /////////////////////////////////////////////////////
         var createRequest = false;
+
+        // barrier is externally-resolved promise
+        var barrier = function() {
```

Comment:
I wonder - can we extract `barrier` out somewhere so it can be reused elsewhere, like in the PPC code?

Reply:
@matt-thomason I can move it to the global scope (this part is isolated, and runs async, waits for requirements) and change its name to something more unique, eg boltBarrier. PPC already uses things from here (global scoped section). However, as far as I am aware, promises are not used in PPC.

Reply:
done